Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -323,51 +323,67 @@ struct ASTopSiteSourceUX {
     static let CellIdentifier = "TopSiteItemCell"
 }
 
-protocol ASHorizontalScrollDelegate {
-    func numberOfVerticalItems(estimatedItems: Int) -> Int
-    func numberOfHorizontalItems(estimatedItems: Int) -> Int
+protocol ASHorizontalLayoutDelegate {
+    func numberOfVerticalItems() -> Int
+    func numberOfHorizontalItems() -> Int
 }
 
 /*
  This Delegate/DataSource is used to manage the ASHorizontalScrollCell's UICollectionView. 
  This is left generic enough for it to be re used for other parts of Activity Stream.
  */
 
-class ASHorizontalScrollSource: NSObject, UICollectionViewDelegate, UICollectionViewDataSource, ASHorizontalScrollDelegate {
+class ASHorizontalScrollCellManager: NSObject, UICollectionViewDelegate, UICollectionViewDataSource, ASHorizontalLayoutDelegate {
+
+    var _content = [TopSiteItem]()
+    var content: [TopSiteItem] {
```

Comment:
This seems misleading to me. When I assign items to `content` I don't expect some of the items to not make it into the list. I can see this being confusing for someone trying to assign to this property. I would try to make the side effect known either by moving the check outside this setter at the call site or creating a method instead of a computed property that expresses what happens.